### PR TITLE
var nullability should be based on initializer

### PR DESCRIPTION
**Pull request by cston**

### CODE REVIEW

File: src/EditorFeatures/CSharpTest/SignatureHelp/InvocationExpressionSignatureHelpProviderTests.cs [hunk 1]
```diff
@@ -522,7 +522,7 @@ static List<T> Goo<T>(T t)
             var expectedOrderedItems = new List<SignatureHelpTestItem>
             {
                 new SignatureHelpTestItem(
-$@"void List<'a?>.Add('a? item)
+$@"void List<'a>.Add('a item)
```
Comment (cston) [inline src/EditorFeatures/CSharpTest/SignatureHelp/InvocationExpressionSignatureHelpProviderTests.cs]: Reverts an earlier change in this branch.

**Comment by cston** [comment]
@dotnet/roslyn-compiler, @jcouv, @alekseyts, please review.

File: src/Compilers/CSharp/Portable/BoundTree/BoundExpressionExtensions.cs [hunk 2]
```diff
@@ -186,7 +186,7 @@ internal static TypeSymbolWithAnnotations GetTypeAndNullability(this BoundExpres
             }
             // PROTOTYPE(NullableReferenceTypes): Could we track nullability always,
             // even in C#7, but only report warnings when the feature is enabled?
-            var isNullable = includeNullability ? expr.IsNullable() : null;
+            var isNullable = includeNullability && !type.IsErrorType()? expr.IsNullable() : null;
```
Comment (jcouv) [inline src/Compilers/CSharp/Portable/BoundTree/BoundExpressionExtensions.cs]: Nit: formatting around question mark

File: src/Compilers/CSharp/Portable/Binder/Binder_Statements.cs [hunk 3]
```diff
@@ -856,8 +856,7 @@ private TypeSymbolWithAnnotations BindVariableType(CSharpSyntaxNode declarationN
                 // If we got a good result then swap the inferred type for the "var" 
                 if ((object)initializerOpt?.Type != null)
                 {
-                    // Default inferred reference types to a nullable state.
-                    declTypeOpt = TypeSymbolWithAnnotations.Create(initializerOpt.Type, isNullableIfReferenceType: declarator.IsFeatureStaticNullCheckingEnabled());
+                    declTypeOpt = initializerOpt.GetTypeAndNullability(declarator.IsFeatureStaticNullCheckingEnabled());
```
Comment (jcouv) [inline src/Compilers/CSharp/Portable/Binder/Binder_Statements.cs]: Later on, we discussed doing binding with `includeNullability: true`. In the meantime, do you have a test with `var` and the feature disabled that looks at the nullability of the local symbol (I expect oblivious).

File: src/Compilers/CSharp/Portable/BoundTree/BoundExpressionExtensions.cs (hunk 2 above)
Comment (cston) [reply src/Compilers/CSharp/Portable/BoundTree/BoundExpressionExtensions.cs]: Fixed.

File: src/Compilers/CSharp/Portable/Binder/Binder_Statements.cs (hunk 3 above)
Comment (cston) [reply src/Compilers/CSharp/Portable/Binder/Binder_Statements.cs]: Good idea. Added.

**Comment by cston** [comment]
@dotnet/roslyn-compiler, @AlekseyTs, please review.

**Comment by AlekseyTs** [comment]
It doesn't look like this PR changes how type is inferred for ```out`` and ```deconstruction``` variables. Consider adding tests and making necessary changes in implementation.

**Pull request merged**